cmd/monaco/dependencygraph/dependencygraph.go

```diff
@@ -74,6 +76,18 @@ func writeGraphFiles(fs afero.Fs, manifestPath string, environmentNames []string
 			return fmt.Errorf("failed to encode dependency graph to DOT for environment %q: %w", e, err)
 		}
 		file := filepath.Join(folderPath, fmt.Sprintf("dependency_graph_%s.dot", e))
+
+		exists, err := afero.Exists(fs, file)
+		if err != nil {
+			return fmt.Errorf("failed to validate if output file %q already exists: %w", file, err)
+		}
+		if exists {
+			time := timeutils.TimeAnchor().Format("20060102-150405")
+			newFile := filepath.Join(folderPath, fmt.Sprintf("dependency_graph_%s_%s.dot", e, time))
+			log.Debug("Output file %q already exists, creating %q instead", file, newFile)
```

Comment:
Should we log the files as structured logs?

Reply:
done

Reply:
Here's a sample error log

```json
{
  "level": "error",
  "ts": "2023-07-24T16:02:43+02:00",
  "msg": "Failed to create dependency graph files: failed to load manifest \"cmd/monaco/dependencygraph/test-resources/manifest.yaml\": encountered multiple errors: [cmd/monaco/dependencygraph/test-resources/manifest.yaml:default:env1: failed to parse auth section: error parsing token: environment-variable \"TOKEN\" was not found, cmd/monaco/dependencygraph/test-resources/manifest.yaml:default:env2: failed to parse auth section: error parsing token: environment-variable \"TOKEN\" was not found]",
  "error": {
    "type": "dependencygraph.ExportError",
    "details": {
      "reason": {
        "errors": [
          {
            "manifestPath": "cmd/monaco/dependencygraph/test-resources/manifest.yaml",
            "reason": "failed to parse auth section: error parsing token: environment-variable \"TOKEN\" was not found",
            "environmentDetails": {
              "group": "default",
              "environment": "env1"
            }
          },
          {
            "manifestPath": "cmd/monaco/dependencygraph/test-resources/manifest.yaml",
            "reason": "failed to parse auth section: error parsing token: environment-variable \"TOKEN\" was not found",
            "environmentDetails": {
              "group": "default",
              "environment": "env2"
            }
          }
        ]
      },
      "manifestFile": "cmd/monaco/dependencygraph/test-resources/manifest.yaml"
    }
  },
  "manifestFile": "cmd/monaco/dependencygraph/test-resources/manifest.yaml",
  "outputFolder": ""
}
```